### PR TITLE
Con 2417 migrate apprentice details

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapperTests.cs [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+        private GetApprenticeshipResponse _apprenticeshipResponse;
+        private GetPriceEpisodesResponse _priceEpisodesResponse;
+        private GetApprenticeshipUpdatesResponse _apprenticeshipUpdatesResponse;
+        private GetDataLocksResponse _dataLocksResponse;
+        private GetChangeOfPartyRequestsResponse _changeOfPartyRequestsResponse;
+        private GetTrainingProgrammeResponse _trainingProgrammeResponse;
+
+        private ApprenticeshipDetailsRequest _request;
+        private ApprenticeshipDetailsRequestToViewModelMapper _mapper;
+
```
Comment (hkf-tech) [inline src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapperTests.cs:37]: This file has extra white space on multiple places

**Review by hkf-tech — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/DataLockStatusExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using SFA.DAS.CommitmentsV2.Types;
+using System.Collections.Generic;
+using System.Linq;
+using static SFA.DAS.CommitmentsV2.Api.Types.Responses.GetDataLocksResponse;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Extensions
+{
+    public static class DataLockStatusExtensions
```
[the remark this answers is not in the document]
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/DataLockStatusExtensions.cs:8]: added

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/ILinkGeneratorExtensions.cs [hunk 3]
```diff
@@ -22,6 +22,63 @@ public static string PayeSchemes(this ILinkGenerator linkGenerator, string accou
             return linkGenerator.CommitmentsLink($"accounts/{accountHashedId}/apprentices/manage/{hashedApprenticeshipId}/details");
         }
 
+        public static string ReviewChanges(
```
[the remark this answers is not in the document]
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/ILinkGeneratorExtensions.cs:25]: done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapper.cs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                    TrainingType = getTrainingProgrammeTask.TrainingProgramme.ProgrammeType,
+                    Cost = priceEpisodes.PriceEpisodes.GetPrice(),
+                    ApprenticeshipStatus = apprenticeship.Status,
+                    Status = statusText,
+                    ProviderName = apprenticeship.ProviderName,
+                    PendingChanges = pendingChange,
+                    EmployerReference = apprenticeship.EmployerReference,
+                    CohortReference = _encodingService.Encode(apprenticeship.CohortId, EncodingType.CohortReference),
+                    EnableEdit = enableEdit,
+                    CanEditStatus = !(new List<ApprenticeshipStatus> { ApprenticeshipStatus.Completed, ApprenticeshipStatus.Stopped }).Contains(apprenticeship.Status),
```
[the remark this answers is not in the document]
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ApprenticeshipDetailsRequestToViewModelMapper.cs]: 👍

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/ILinkGeneratorExtensions.cs [hunk 5]
```diff
@@ -22,6 +22,63 @@ public static string PayeSchemes(this ILinkGenerator linkGenerator, string accou
...
+            return linkGenerator.CommitmentsLink($"accounts/{accountHashedId}/apprentices/manage/{hashedApprenticeshipId}/edit");
+        }
+
+
+        public static string ChangeApprenticeStatus(
+             this ILinkGenerator linkGenerator,
+             string accountHashedId,
+             string hashedApprenticeshipId)
+        {
+            return linkGenerator.CommitmentsLink($"accounts/{accountHashedId}/apprentices/manage/{hashedApprenticeshipId}/details/statuschange");
```
[the remark this answers is not in the document]
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/ILinkGeneratorExtensions.cs]: done

**Review by hkf-tech — APPROVED** [review]

**Pull request merged**